include/envoy/upstream/load_balancer_type.h

```diff
@@ -25,6 +25,14 @@ enum class LoadBalancerType {
   ClusterProvided
 };
 
+struct SubsetSelector {
+  std::set<std::string> selector_keys;
+  envoy::api::v2::Cluster::LbSubsetConfig::LbSubsetSelector::LbSubsetSelectorFallbackPolicy
+      fallback_policy;
+};
+
+typedef std::shared_ptr<SubsetSelector> SubsetSelectorPtr;
```

Comment:
Nit: most code is now preferring `using` to `typedef` for this.

Reply:
that's a new keyword i was not aware of :)